File: Programming EF Code First/chapter6/Chapter6/Chapter 6 VB/DataAnnotations/Model/PersonPhoto.vb
Imports System.ComponentModel.DataAnnotations

Namespace Model
  <Table("People")>
  Public Class PersonPhoto
	<Key, ForeignKey("PhotoOf")>
	Public Property PersonId() As Integer
	<Column(TypeName := "image")>
	Public Property Photo() As Byte()
	Public Property Caption() As String

	Public Property PhotoOf() As Person
  End Class
End Namespace

File: Programing EF DbContext/AfterCompletingChapter6/Model/PersonPhoto.cs
using System.ComponentModel.DataAnnotations;

namespace Model
{
  [Table("People")]
  public class PersonPhoto
  {
    [Key]
    [ForeignKey("PhotoOf")]
    public int PersonId { get; set; }
    [Column(TypeName = "image")]
    public byte[] Photo { get; set; }
    public string Caption { get; set; }

    public Person PhotoOf { get; set; }
  }
}